editor/code_editor.cpp

```diff
@@ -1633,80 +1600,61 @@ void CodeTextEditor::duplicate_selection() {
 void CodeTextEditor::toggle_inline_comment(const String &delimiter) {
 	text_editor->begin_complex_operation();
 
+	// Comment only if there's any uncommented lines in selection.
+	Vector<int> lines = _get_affected_lines();
+	bool is_commented = true;
+	for (const int &line : lines) {
+		if (!text_editor->get_line(line).begins_with(delimiter)) {
+			is_commented = false;
```

Comment:
Ditto here I guess.

Reply:
I'm not sure what are you referring here...

Reply:
My Bad. I was referring to bookmarks comment, VSCode treats toggling comments per selection / caret, however JetBrains (PyCharm) does not.

In both cases if they act inversely to this. So if there is a single un-commented line it will comment that line. Rather then here, where if there is a single commented like it will un-comment it. Whether that is the approach we should take.

Reply:
> In both cases if they act inversely to this. So if there is a single un-commented line it will comment that line.

Umm, my version does it also this way, and yes, that's the way it should be.
I often have some blocks commented out, and then might want to temporarily comment out the surrounding block as well. With current behavior it's non-destructive operation, toggling it twice in a row is quaranteed to return you into original state. Plus, if you have some actual comments in there, how we could detect them and not toggle?

As for toggle per selection vs all selections, yes, that seems to be the common case. I can change it, however then we're back to square one. If 2 selections overlap a single line, should it be subjected only to topmost selection then (as it covers the beginning of the line, where the delimiter is)? That seems to be VS Code way of doing things.

Reply:
My bad for misreading the code. It does seem to be more common to treat each caret separately so I think going with that and copying VSCode's behaviour sounds good to me.